From: Jan Kara <jack@suse.cz>
Subject: Update readahead tunable
References: bnc#548529
Patch-mainline: no

Increase read_ahead_kb to values from SLES10 SP3 to get
back sequential IO performance if we are not compiling a -desktop kernel
flavor.

Update jeffm 21 Dec 2014:
- BLK_DEF_MAX_SECTORS was removed in 3.19-rc1.
- read_ahead_kb really should be a sysctl, but systemd magic needs to
  happen to get the ordering right. Otherwise, it'll be applied after
  the local file systems are mounted and it'll have no effect.

Signed-off-by: Jan Kara <jack@suse.cz>

---
 include/linux/blkdev.h |    4 ++++
 include/linux/mm.h     |    4 ++++
 2 files changed, 8 insertions(+)

--- a/include/linux/mm.h
+++ b/include/linux/mm.h
@@ -1307,7 +1307,11 @@ int write_one_page(struct page *page, in
 void task_dirty_inc(struct task_struct *tsk);
 
 /* readahead.c */
+#ifndef CONFIG_KERNEL_DESKTOP
+#define VM_MAX_READAHEAD	512	/* kbytes */
+#else
 #define VM_MAX_READAHEAD	128	/* kbytes */
+#endif
 #define VM_MIN_READAHEAD	16	/* kbytes (includes current page) */
 
 int force_page_cache_readahead(struct address_space *mapping, struct file *filp,
